### PR TITLE
GTK Reflashing & Write Cancellation

### LINKED ISSUE #51: Feature request: Add ability to continue flashing different images after first one is complete
**Distribution (run `cat /etc/os-release`):**
 Pop!_OS 18.04
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
popsicle-gtk

**Issue/Bug Description:**
As someone who uses popsicle quite a lot to image a number of different images, after writing an image to an external drive, I currently have to close the application and then relaunch it to image another drive with a different image. 

I can imagine a button in the top left (opposite from the "Close" button) that would go back to the "choose an image" page.

### LINKED ISSUE #52: Popsicle freezes when clicking next with image selected
**Distribution (run `cat /etc/os-release`):**

 aaronhoneycutt@pop-os:~/Downloads$ cat /etc/os-release
NAME="Pop!_OS"
VERSION="18.04 LTS"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 18.04 LTS"
VERSION_ID="18.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

aaronhoneycutt@pop-os:~/Downloads$ apt policy popsicle-gtk 
popsicle-gtk:
  Installed: 0.1.5~1530889767~18.04~2ba9db3
  Candidate: 0.1.5~1530889767~18.04~2ba9db3
  Version table:
 *** 0.1.5~1530889767~18.04~2ba9db3 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status

**Issue/Bug Description:**

The application freezes after an image is selected and then clicking Next

**Steps to reproduce (if you know):**

**Expected behavior:**

**Other Notes:**

[log.txt](https://github.com/pop-os/popsicle/files/2202520/log.txt)

### PR DESCRIPTION
**Pull request by mmstick**
- libpopsicle was modified to enable early killing of the device flashing process
- state is reset when switching back to the main view in popsicle-gtk
- popsicle-gtk will revert back to the main view when:
  - Cancel is selected on the device flashing view
  - Back is selected on the summary view
- popsicle-gtk GTK widgets are now Rc'd together

Closes #51

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,10 @@
 name = "popsicle"
 description = "USB Flasher"
 version = "0.1.5"
-authors = ["Jeremy Soller <jeremy@system76.com>"]
+authors = [
+    "Jeremy Soller <jeremy@system76.com>",
+    "Michael Aaron Murphy <michael@system76.com>",
```
Comment (brs17) [inline Cargo.toml]: It's about time! :tada:

**Comment by brs17** [comment]
@cassidyjames Asked for your review just cause there are some graphical changes (and it never hurts).

**Comment by mmstick** [comment]
@cassidyjames Unless the user is not changing the image, and immediately progressing back to device selection, since it is previous image is still selected at that point. In that case, should it be "Flash Again"?

**Comment by cassidyjames** [comment]
@mmstick that could work

**Comment by brs17** [comment]
When canceling a write midway through, the `Cancel` button remains highlighted red even when back at the first screen.
aftercancel

 I would imagine this should dehighlight (to look the way it does originally on the first screen).
firstpage

That also makes me wonder/think if that `Cancel` button on the first screen should say `Close` like on the last screen (but on the left side).
flashagain


Thoughts @cassidyjames ?

**Comment by brs17** [comment]
Functionality wise, it looks good though.

Just waiting for that last commit to build...

**Comment by cassidyjames** [comment]
It would also be great to update screenshots in the repo with this PR so we keep those up to date with the current UI.

**Comment by mmstick** [comment]
The summary view now says "Done", and I've fixed the next button on the device selection view. Will add new screenshots.

**Comment by mmstick** [comment]
Found an issue that can cause the device selection view to lock if a device is selected and then any device is removed. Looking for the cause of the lock.

**Comment by mmstick** [comment]
Screenshots are now updated.

**Comment by mmstick** [comment]
[Rendered](https://github.com/pop-os/popsicle/blob/reflashing/README.md)

**Pull request merged**